pkg/storageincentives/staking/contract.go

```diff
@@ -313,3 +318,33 @@ func (c *contract) paused(ctx context.Context) (bool, error) {
 
 	return results[0].(bool), nil
 }
+
+func (c *contract) IsOverlayFrozen(ctx context.Context) (bool, error) {
+	callData, err := c.stakingContractABI.Pack("overlayNotFrozen")
+	if err != nil {
+		return false, err
+	}
+
+	result, err := c.transactionService.Call(ctx, &transaction.TxRequest{
+		To:   &c.stakingContractAddress,
+		Data: callData,
+	})
+	if err != nil {
+		return false, err
+	}
+
+	results, err := c.stakingContractABI.Unpack("overlayNotFrozen", result)
+	if err != nil {
+		return false, err
+	}
+
+	if len(results) == 0 {
+		return false, errors.New("unexpected empty results")
+	}
+	// true if overlay is not frozen
+	if results[0].(bool) {
+		return false, nil
+	}
+	// false if overlay is frozen
+	return true, nil
```

Comment:
this logic here seems inverted?

`if results[0].(bool)` then it should return true, but in code it returns false?

Reply:
Yes, this is inverted.
https://github.com/ethersphere/storage-incentives/blob/04be7ba384d77e60b2dc99e7c679bf98d529f052/src/Staking.sol#L76

Contract Logic
true  -> Node not frozen
false -> Node is frozen

`IsOverlayFrozen` is the opposite of a contract.

Therefore `if results[0].(bool)` true would be false. Means the overlay is not frozen.

Reply:
would it be more readable if this part is written like so:

```
isNotFrozen := results[0].(bool)
return !isNotFrozen, nil
```

Reply:
This would be nice.